### PR TITLE
feat: Added basic observability service implementation

### PR DESCRIPTION
**Pull request by JigarJoshi**
TODO:

- [ ] Add namespace tag to scope metrics query.
- [ ] Configure allowed queries.

### CODE REVIEW

File: server/services/v1/observability.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to query metrics: reason = "+err.Error())
+	}
+
+	bodyStr := string(body)
+	if resp.StatusCode == 200 {
+		var ddResp MetricsQueryResp
+		err = json.Unmarshal(body, &ddResp)
```
Comment (efirs) [inline server/services/v1/observability.go:97]: add error check.

File: server/services/v1/observability.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+		var dataPoints = make([]*api.DataPoint, len(ddResp.Series[0].PointList))
```
Comment (efirs) [inline server/services/v1/observability.go:98]: check that there is atleast one series in the response before accessing it.

File: server/services/v1/observability.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		for i := range ddResp.Series[0].PointList {
+			dataPoints[i] = &api.DataPoint{}
+			dataPoints[i].Timestamp = int64(ddResp.Series[0].PointList[i][0])
```
Comment (efirs) [inline server/services/v1/observability.go:101]: Do we need to set value too?

File: server/services/v1/observability.go (hunk 1 above)
Comment (JigarJoshi) [reply server/services/v1/observability.go:97]: fixed.

File: server/services/v1/observability.go (hunk 2 above)
Comment (JigarJoshi) [reply server/services/v1/observability.go]: added check.

File: server/services/v1/observability.go (hunk 3 above)
Comment (JigarJoshi) [reply server/services/v1/observability.go]: yes it was totally missed. thanks.

File: server/services/v1/observability.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+type Datadog struct {
+	config.ObservabilityConfig
+}
+
+type MetricsQueryResp struct {
+	FromDate int64 `json:"from_date"`
+	ToDate   int64 `json:"to_date"`
+	Query    string
+	Series   []struct {
+		PointList [][]float64 `json:"pointlist"`
```
Comment (efirs) [inline server/services/v1/observability.go:60]: Can we declare it like below? So we can remove size checks after unmarshaling:
```
Series [1]struct {
		PointList [][2]float64 `json:"pointlist"`
```

Comment (JigarJoshi) [reply server/services/v1/observability.go:60]: yes definitely.

**Comment by JigarJoshi** [comment]
abandoning this PR. I will create new one.

**Pull request closed without merging**